### ISSUE TITLE
npm install fails with 'invalid version: 3001.0001.0000-dev-harmony-fb'

### ISSUE THREAD

**Issue by jimlloyd** [facebook/esprima#105]
A project I am working on can no longer do a fresh npm install. The error I get is:

```
npm ERR! Darwin 14.1.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v0.12.0
npm ERR! npm  v2.5.1

npm ERR! invalid version: 3001.0001.0000-dev-harmony-fb
```

I don't directly install esprima-fb, and am still trying to figure out how it is indirectly installed. The npm-debug.log is 20K lines long. I've included it in this gist: https://gist.github.com/jimlloyd/3e61c65e71c4537cfae8

**Comment by zpao** [comment]
We resolved this long ago. You'll probably want to figure out which package is depending on that old version and get the author to upgrade.

**Comment by jimlloyd** [comment]
I didn't install astw directly. I see that astw is dependent on that exact version of esprima-fb, but astw hasn't change since since March 20, 2014, nearly a year ago. My project is relatively new. I started development about 3 months ago. It has built fine until today. Since astw hasn't change in a year, I can only guess that a recent change in some other module that I indirectly depend on has caused this problem with astw to surface.

**Comment by jimlloyd** [comment]
I want to leave a final note here in case others have this problem in the future.

It appears to me that there is some other failure mode that has nothing to do with the bug in esprima-fb that was resolved long ago. I'm pretty sure there was some environmental problem causing my problem yesterday, likely on my own machine, though possibly elsewhere in the ecosystem. I suspected this before logging the issue, and tried to clear it using `npm cache clear` and even `npm cache clear -f`, but neither of these fixed the problem. I use the node version manager [n](https://www.npmjs.com/package/n) to keep multiple versions of node installed. I had already tried switching from 0.12 to 0.11.15,  which didn't resolve the problem. But eventually I switched to 0.10.35, and discovered that I could install. I switched back to 0.12, and found the problem was resolved with that version as well. I conjecture that since moves files around, it resolved a subtle problem with a corrupted file or file permissions.

It is possible that some other change in the npm package ecosystem occurred in the meantime, but I doubt it. In fact, my project still uses esprima-fb@3001.1.0-dev-harmony-fb:

```
$ npm ls | grep --context=2 esprima-fb
│ │ │ ├─┬ lexical-scope@1.1.0
│ │ │ │ └─┬ astw@1.1.0
│ │ │ │   └── esprima-fb@3001.1.0-dev-harmony-fb
```
